youtube_dl/postprocessor/metadatafromtitle.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+
+import re
+
+from .common import PostProcessor
+from ..utils import PostProcessingError
+
+
+class MetadataFromTitlePPError(PostProcessingError):
+    pass
+
+
+class MetadataFromTitlePP(PostProcessor):
+    def __init__(self, downloader, titleformat):
+        self._titleformat = titleformat
+        self._titleregex, self._attributes = self.fmtToRegex(titleformat)
+
+    def fmtToRegex(self, fmt):
```

Comment:
We should probably add some tests for the method, otherwise it's harder to edit it.